code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -412,55 +412,65 @@
 	if(should_stun)
 		Paralyze(60)
 
-/mob/living/carbon/proc/help_shake_act(mob/living/carbon/M)
-	if(on_fire)
-		to_chat(M, "<span class='warning'>You can't put [p_them()] out with just your bare hands!</span>")
-		return
+/mob/living/carbon/proc/help_shake_act(mob/living/L)
```

Comment:
Better variable names.

Also, documentation. I know you didn't make this proc, but while you're here.

Reply:
So even though already I went through this proc and replaced all of the "M"s in it with "L"s and "C"s (which most other procs use to refer to living mobs and carbons), I now have to go though it AGAIN to replace those "L"s with "Living"s and those "C"s with "Carbon"s?

Reply:
It's something that we're slowly weeding out, yeah.

Reply:
we are? i thought it was a fairly intuitive standard

Reply:
It's a somewhat common standard in other codebases, and I've had it expected of PRs of mine.

The idea behind it is not just that things like `C` and `M` can apply to multiple things, but also that even things like `carbon` are vague. What's the relevance of the carbon here? Is it the user? The target? The victim? Etc.

Reply:
Using it directly as a function argument does lead to potentially confusing code, yeah, i didn't see the context.

The way i've used it so far is when a type has to be casted as a more specific type for a limited time, and all the code related would be relegated to the if block that checked the type
```
if(istype(enemy_mob, mob/living/carbon)
	var/mob/living/carbon/C = enemy_mob
	C.thing()
	etc...
```
That way there is no ambiguity between the sources in most cases, and where there is we can still use full names.
As for ambiguity with other types with the same initials, i've seen double initials for the less common one if possible (e.g. C = carbon, CT = coal_thrower).
I'd kinda prefer to keep this usage alive, since it makes it evident there's been a typecasting at some point.

Reply:
^

Reply:
according to my reasoning above, i would name the argument here something like `target`, since it's not an actual typecasting.

Reply:
You're not always looking in the same place as the declaration--`enemy_carbon` is a better name in your sample than `C`.

Reply:
That also works, but it gets kinda long (e.g. it could change C => teleport_target_carbon, AM => teleport_target_atom_movable, and so on), and seems unnecessary in most cases, which often are limited to one or few lines of code that require the typecasting.
If we _have_ to stick to a rigid standard then it's fine, but i think that initials for short segments and full rename for long segments should work pretty well.